### PR TITLE
Configure DtCalib AlCaReco for heavy Ion 2018

### PR DESCRIPTION
**Pull request by tocheng**
Greetings,

The PR is to add DtCalibHI into the AlCaReco stream.
The DtCalib in the AlCaReco stream is replaced by sequence in DtCalibHI using the pp_on_AA_2018 era.
Also the DtCalibOutput is replaced by the DtCalibHIOut using the pp_on_AA_2018 era.

The configuration is checked using 
python $CMSSW_BASE/src/Configuration/DataProcessing/test/RunExpressProcessing.py --scenario ppEra_Run2_2018_pp_on_AA --global-tag 103X_dataRun2_Express_v1 --lfn /store/whatever --fevt --dqmio  --alcareco DtCalib

After changing pickle to python config,
one can see that the DtCalib path and output is replaced  by the path and output in DtCalibHI

### CODE REVIEW

**Comment by tocheng** [comment]
@cmsbuild please test

File: CalibMuon/DTCalibration/python/ALCARECODtCalib_Output_cff.py [hunk 1]
```diff
@@ -40,3 +40,13 @@
 import copy
 OutALCARECODtCalib = copy.deepcopy(OutALCARECODtCalib_noDrop)
 OutALCARECODtCalib.outputCommands.insert(0, "drop *")
+
+## customizations for the pp_on_AA_2018 eras
+from Configuration.Eras.Modifier_pp_on_AA_2018_cff import pp_on_AA_2018
+
+OutALCARECODtCalibHI = copy.deepcopy(OutALCARECODtCalib_noDrop)
+OutALCARECODtCalibHI.outputCommands.insert(0, "drop *")
+OutALCARECODtCalibHI.outputCommands.append("keep *_hiSelectedVertex_*_*")
```
Comment (mmusich) [inline CalibMuon/DTCalibration/python/ALCARECODtCalib_Output_cff.py]: Hi @tocheng I am wondering is this event product available with the`pp_on_AA_2018` era customisations?
@mandrenguyen

Comment (tocheng) [reply CalibMuon/DTCalibration/python/ALCARECODtCalib_Output_cff.py]: @mmusich Thanks for pointing it out. I checked the configuration generated using DP config with pp_on_AA scenario, the hiSelectedVertex is not in the sequence. It should be the module in heavyIon typed reconstruction. I can remove it after the test is done.

**Comment by mmusich** [comment]
Hello @fabiocos, @slava77,
while this PR runs fine in the last IB of 10.3.X using:
```bash
python $CMSSW_BASE/src/Configuration/DataProcessing/test/RunExpressProcessing.py --scenario ppEra_Run2_2018_pp_on_AA --global-tag 103X_dataRun2_Express_v1 --lfn /store/t0streamer/Data/HIExpress/000/325/112/run325112_ls0013_streamHIExpress_StorageManager.dat --fevt --dqmio --alcareco DtCalib
```
and then changing the the event source with:
```
process.source = cms.Source("NewEventStreamFileReader"
```

while doing the same in 10.4.X I notice I incur in this error:
```
----- Begin Fatal Exception 25-Oct-2018 22:56:57 CEST-----------------------
An exception of category 'ScheduleExecutionFailure' occurred while
   [0] Calling beginJob
Exception Message:
Unrunnable schedule
Module run order problem found: 
hiConformalPixelTracks after hiConformalPixelTracksPhase1Filter [path reconstruction_step], hiConformalPixelTracksPhase1Filter after pixelFitterByConformalMappingAndLine [path reconstruction_step], pixelFitterByConformalMappingAndLine after hiConformalPixelTracksPhase1HitQuadrupletsCA [path reconstruction_step], hiConformalPixelTracksPhase1HitQuadrupletsCA after hiConformalPixelTracksPhase1HitDoubletsCA [path reconstruction_step], hiConformalPixelTracksPhase1HitDoubletsCA after hiConformalPixelTracksPhase1SeedLayers [path reconstruction_step], hiConformalPixelTracksPhase1SeedLayers after hiConformalPixelTracksPhase1TrackingRegions [path reconstruction_step], hiConformalPixelTracksPhase1TrackingRegions after hiClusterCompatibility [path reconstruction_step], hiClusterCompatibility after hiCentrality [path reconstruction_step], hiCentrality consumes hiConformalPixelTracks
 Running in the threaded framework would lead to indeterminate results.
 Please change order of modules in mentioned Path(s) to avoid inconsistent module ordering.
----- End Fatal Exception -------------------------------------------------
```

the same error is also obtained in a vanilla `CMSSW_10_4_X_2018-10-25-1100`. 
Is there any substantial difference between the tip of 10.3.X and the tip of 10.4.X that can cause this?

**Comment by mandrenguyen** [comment]
No, hiSelectedVertex is not available.
It’s from the old HeavyIons scenario.
With the pp_on_AA era, we use offlinePrimaryVertices

**Comment by tocheng** [comment]
@cmsbuild please test

**Comment by tocheng** [comment]
+1

**Comment by fabiocos** [comment]
@tocheng did you verify it works? If I use your recipe

python $CMSSW_BASE/src/Configuration/DataProcessing/test/RunExpressProcessing.py --scenario ppEra_Run2_2018_pp_on_AA --global-tag 103X_dataRun2_Express_v1 --lfn /store/whatever --fevt --dqmio --alcareco DtCalib

and replace the input with 

process.source = cms.Source("NewEventStreamFileReader",
fileNames = cms.untracked.vstring('/store/t0streamer/Data/HIPhysicsCommissioning/000/325/112/run325112_ls0006_streamHIPhysicsCommissioning_StorageManage\
r.dat')
)

I get the following problem:

----- Begin Fatal Exception 26-Oct-2018 06:44:50 CEST-----------------------
An exception of category 'ScheduleExecutionFailure' occurred while
   [0] Calling beginJob
Exception Message:
Unrunnable schedule
Module run order problem found: 
hiConformalPixelTracks after hiConformalPixelTracksPhase1Filter [path reconstruction_step], hiConformalPixelTracksPhase1Filter after pixelFitterByConformalMappingAndLine [path reconstruction_step], pixelFitterByConformalMappingAndLine after hiConformalPixelTracksPhase1HitQuadrupletsCA [path reconstruction_step], hiConformalPixelTracksPhase1HitQuadrupletsCA after hiConformalPixelTracksPhase1HitDoubletsCA [path reconstruction_step], hiConformalPixelTracksPhase1HitDoubletsCA after hiConformalPixelTracksPhase1SeedLayers [path reconstruction_step], hiConformalPixelTracksPhase1SeedLayers after hiConformalPixelTracksPhase1TrackingRegions [path reconstruction_step], hiConformalPixelTracksPhase1TrackingRegions after hiClusterCompatibility [path reconstruction_step], hiClusterCompatibility after hiCentrality [path reconstruction_step], hiCentrality consumes hiConformalPixelTracks
 Running in the threaded framework would lead to indeterminate results.
 Please change order of modules in mentioned Path(s) to avoid inconsistent module ordering.
----- End Fatal Exception -------------------------------------------------

**Comment by tocheng** [comment]
@fabiocos 
Marco did some test. The error has nothing to do with PR and doesn't appear in 10_3_X
> Hello @fabiocos, @slava77,
> while this PR runs fine in the last IB of 10.3.X using:
> 
> ```shell
> python $CMSSW_BASE/src/Configuration/DataProcessing/test/RunExpressProcessing.py --scenario ppEra_Run2_2018_pp_on_AA --global-tag 103X_dataRun2_Express_v1 --lfn /store/t0streamer/Data/HIExpress/000/325/112/run325112_ls0013_streamHIExpress_StorageManager.dat --fevt --dqmio --alcareco DtCalib
> ```
> 
> and then changing the event source with:
> 
> ```
> process.source = cms.Source("NewEventStreamFileReader"
> ```
> 
> while doing the same in 10.4.X I notice I incur in this error:
> 
> ```
> ----- Begin Fatal Exception 25-Oct-2018 22:56:57 CEST-----------------------
> An exception of category 'ScheduleExecutionFailure' occurred while
>    [0] Calling beginJob
> Exception Message:
> Unrunnable schedule
> Module run order problem found: 
> hiConformalPixelTracks after hiConformalPixelTracksPhase1Filter [path reconstruction_step], hiConformalPixelTracksPhase1Filter after pixelFitterByConformalMappingAndLine [path reconstruction_step], pixelFitterByConformalMappingAndLine after hiConformalPixelTracksPhase1HitQuadrupletsCA [path reconstruction_step], hiConformalPixelTracksPhase1HitQuadrupletsCA after hiConformalPixelTracksPhase1HitDoubletsCA [path reconstruction_step], hiConformalPixelTracksPhase1HitDoubletsCA after hiConformalPixelTracksPhase1SeedLayers [path reconstruction_step], hiConformalPixelTracksPhase1SeedLayers after hiConformalPixelTracksPhase1TrackingRegions [path reconstruction_step], hiConformalPixelTracksPhase1TrackingRegions after hiClusterCompatibility [path reconstruction_step], hiClusterCompatibility after hiCentrality [path reconstruction_step], hiCentrality consumes hiConformalPixelTracks
>  Running in the threaded framework would lead to indeterminate results.
>  Please change order of modules in mentioned Path(s) to avoid inconsistent module ordering.
> ----- End Fatal Exception -------------------------------------------------
> ```
> 
> the same error is also obtained in a vanilla `CMSSW_10_4_X_2018-10-25-1100`.
> Is there any substantial difference between the tip of 10.3.X and the tip of 10.4.X that can cause this?

**Comment by fabiocos** [comment]
ok, I confirm that with the latest fix of @slava77 #25011 the PR looks to work

**Comment by fabiocos** [comment]
+1

**Pull request merged**